compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2712,11 +2715,34 @@ public void visit(BLangQueryExpr queryExpr) {
                         this.dlog.error(collection.pos, DiagnosticCode.NOT_ALLOWED_STREAM_USAGE_WITH_FROM);
                     }
                 }
+            } else if (clause.getKind() == NodeKind.ORDER_BY) {
+                checkOrderFieldValidity(queryExpr.type, (BLangOrderByClause) clause);
             }
             analyzeNode(clause, env);
         }
     }
 
+    private void checkOrderFieldValidity(BType resultType, BLangOrderByClause clause) {
+        resultType = types.resolveExprType(resultType);
+        if (resultType.tag == TypeTags.RECORD) {
```

Comment:
shall we invert the condition and do an early-exit? That way we can avoid one level of right-indentations.
eg:
```java
if (resultType.tag != TypeTags.RECORD) {
    return;
}
```